Add breaking changes to release notes (#2936)
elastic/elasticsearch-py: docs/release-notes/breaking-changes.md

% **Action**<br> Steps for mitigating deprecation impact.
% ::::

## 9.0.0 [elasticsearch-python-client-900-breaking-changes]

> ::::{dropdown} Remove deprecated `Elasticsearch()` options

Reviewer: Turns out this is by design -- formatting isn't support in dropdown titles. see https://github.com/elastic/docs-builder/issues/306#issuecomment-2648514283

> ::::{dropdown} Remove deprecated `Elasticsearch()` options

Author: Thanks for checking, removing them is OK.

For more information, check [PR #2840](https://github.com/elastic/elasticsearch-py/pull/2840).
**Impact**<br> These parameters were removed in favor of more descriptive versions. Using any of those parameters will prevent instantiating the Elasticsearch client.
**Action**<br> Those parameters can be replaced as follows:
 * `timeout` is now `request_timeout`
 * `randomize_hosts` is now `randomize_nodes_in_pool`
 * `host_info_callback` is now `sniffed_node_callback`
 * `sniffer_timeout` is now `min_delay_between_sniffing`
 * `sniff_on_connection_fail` is now `sniff_on_node_failure`
 * `maxsize` is now `connection_per_node`
::::

::::{dropdown} Remove deprecated `url_prefix` and `use_ssl` host keys
When instantiating a new client, `hosts` can be specified as a dictionary. The `url_prefix` and `use_ssl` keys are no longer allowed.
For more information, check [PR #2797](https://github.com/elastic/elasticsearch-py/pull/2797).

**Impact**<br>
Using any of those parameters will prevent instantiating the Elasticsearch client.

**Action**<br>
The parameters can be replaced as follows:
 * `use_ssl` isn't needed as a scheme is required since elasticsearch-py 8.0 (`http` or `https`)

>  * `url_prefix` should be replaced with `path_prefix`, which is more descriptive. Indeed, this functionality allows deploying Elasticsearch under a specific path, such as `http://host:port/path/to/elasticsearch`, instead of the default root path (`http://host:port/`)

Reviewer: ```suggestion
 * `url_prefix` should be replaced with `path_prefix`, which is more descriptive. This functionality allows you to deploy Elasticsearch under a specific path, such as `http://host:port/path/to/elasticsearch`, instead of the default root path (`http://host:port/`)
```

::::

> ::::{dropdown} Removed APIs

Reviewer: ```suggestion
::::{dropdown} Remove APIs
```

Reviewer: Up to you whether to use past tense "Removed" in all of them (I have a slight preference for past tense) -- but the first two BCs use present tense "Remove" (as do all the release notes) so I changed this one to be consistent.

Author: I think I also like the past tense, but we've used the present tense for years now, indeed, so thanks for keeping it consistent.

Elasticsearch 9 removed the kNN search and Unfreeze index APIs.

**Action**<br>
 * The kNN search API has been replaced by the `knn` option in the search API since Elasticsearch 8.4.

>  * The Unfreeze index API was deprecated in Elasticsearch 7.14, and has been removed in Elasticsearch 9 since it is not possible to freeze an index since Elasticsearch 8.0.

Reviewer: ```suggestion
 * The Unfreeze index API was deprecated in Elasticsearch 7.14 and has been removed in Elasticsearch 9.
```

Reviewer: The 8.0 info seems unnecessary to me, but feel free to ignore this change
